exampleCourse/questions/template/checkbox/feedback/server.py

```diff
@@ -0,0 +1,34 @@
+MAX_NUMBER = 20
+
+# For the same of this example, we use a hardcoded list of all primes up to 20.
+# Update this list if you change `MAX_NUMBER` above.
+PRIME_NUMBERS_UNDER_20 = {2, 3, 5, 7, 11, 13, 17, 19}
+
+
+def factors(n):
+    factors = set()
+    for i in range(1, n + 1):
+        if n % i == 0:
+            factors.add(i)
+    return factors
+
+
+def feedback_for_number(number):
+    if number in PRIME_NUMBERS_UNDER_20:
+        return f"Correct! {number} is a prime number."
+    elif number == 1:
+        return "Incorrect; 1 is neither prime nor composite."
+    else:
+        factors_hint = ", ".join(str(factor) for factor in factors(number))
+        return f"Incorrect; {number} has the following factors: {factors_hint}"
```

Comment:
I like the question, but I fear this might be an example where a non-programmer would find it much clearer to just write out all the possible answers and feedbacks in `question.html`. It would even be shorter overall! 😂

I do appreciate that at least you hardcoded the list of primes.

That being said, I haven't thought of a great alternative question to demonstrate computing feedback for checkboxes. For multiple choice a possibility is something like (testing unit conversions):

> If a car travels {{params.distance1}} m in {{params.time1}} minutes, how many hours would it take to travel {{params.distance2}} km at the same speed?

```
distance1 = randint(2,6) * 100 # in meters
distance2 = distance1 / 100 # in kilometers
time1 = randint(2,6) * 10 # in minutes

correct_answer = time1 * 10 / 60 # in hours
distractor1 = time1 * 10 # forgot the conversion to hours
distractor2 = time1 / 100 / 60 # forgot the conversion to km
distractor3 = time1 / 100 # forgot both conversions
distractor4 = time1 * 2 # no particular error, just another value
```

```
<pl-checkbox
   answers-name="numbers"
   number-answers="7"
   fixed-order="true"
   partial-credit="true"
   min-correct="2"
 >
  <pl-answer correct="true">{{params.correct_answer}}</pl-answer>
  <pl-answer correct="false" feedback="remember that the answer should be expressed in hours, not minutes">{{params.distractor1}}</pl-answer>
  <pl-answer correct="false" feedback="note that the second distance is given in kilometers, not meters">{{params.distractor2}}</pl-answer>
  <pl-answer correct="false" feedback="pay attention to all the different units that are being used">{{params.distractor3}}</pl-answer>
  <pl-answer correct="false" feedback="you are not using the correct formula">{{params.distractor4}}</pl-answer>
</pl-checkbox>
```

Reply:
Just pinging on this PR -- I wasn't sure whether you were wanting to address this point still?

Reply:
Yes, I think this is worth addressing. I agree I don't have any great suggestions for _dynamic_ feedback on `<pl-checkbox>`. Maybe we fall back to just showing how to hardcode answer feedback?

I was brainstorming with ChatGPT now and maybe this is an interesting question that would work here: "Which of the following cities are in [country]?" The feedback itself wouldn't be dynamic ("Paris is in France", "Berlin is in Germany", etc.), but it could show how to randomly vary the question, options, and which options are actually correct vs. distractors.

Reply:
I like it!

Reply:
I updated the question to use countries/cities in c0abb65.